packages/nc-gui/composables/useMapViewDataStore.ts

```diff
@@ -0,0 +1,162 @@
+import { reactive } from 'vue'
+import type { ComputedRef, Ref } from 'vue'
+import type { ColumnType, MapType, PaginatedType, TableType, ViewType } from 'nocodb-sdk'
+import { ref, useInjectionState, useMetas } from '#imports'
+import type { Row } from '~/lib'
+
+export const geodataToggleState = reactive({ show: false })
+
+const formatData = (list: Row[]) =>
+  list.map((row) => ({
+    row: { ...row },
+    oldRow: { ...row },
+    rowMeta: {},
+  }))
+
+const [useProvideMapViewStore, useMapViewStore] = useInjectionState(
+  (
+    meta: Ref<TableType | undefined>,
+    viewMeta: Ref<ViewType | MapType | undefined> | ComputedRef<(ViewType & { id: string }) | undefined>,
+    where?: ComputedRef<string | undefined>,
+  ) => {
+    const formattedData = ref<Row[]>([])
+
+    const { api } = useApi()
```

Comment:
maybe also more a question to the NocoDB team: what are the conventions about when to use `useApi` and when `const { $api } = useNuxtApp()`. My understanding is that when calling useApi, it creates a new instance (if it's not called with the parameter `useGlobalInstance=true`) - but in which cases are new instances for the API really necessary? 

In this case I think you can quickly validate/test the assumption that we can just use `$api` (and get rid of the additional `useApi` call) by changing the code and testing that the map view still works as expected when showing, adding, editing and removing geo data.

Reply:
> My understanding is that when calling useApi, it creates a new instance (if it's not called with the parameter useGlobalInstance=true) - but in which cases are new instances for the API really necessary?

Both would work. With `useApi`, it creates a new clone and you can get the loading status and the error status from the corresponding instance.